app-ios/Modules/Sources/FloorMap/SideEventRow.swift

```diff
@@ -0,0 +1,109 @@
+import Assets
+import Component
+import shared
+import SwiftUI
+import Theme
+
+struct SideEventRow: View {
+    var sideEvent: SideEvent
+
+    var body: some View {
+        HStack(alignment: .top, spacing: 12) {
+            VStack(alignment: .leading, spacing: 8) {
+                VStack(alignment: .leading, spacing: 4) {
+                    // Icon & Title
+                    HStack(alignment: .top, spacing: 8) {
+                        sideEvent.icon
+                            .renderingMode(.template)
+                            .frame(width: 16, height: 16)
+                            .foregroundStyle(sideEvent.iconColor)
+
+                        Text(sideEvent.title.currentLangTitle)
+                            .font(.system(size: 14, weight: .bold))
+                    }
+                    // Description
+                    Text(sideEvent.description_.currentLangTitle)
+                        .font(.system(size: 14, weight: .medium))
+                }
+                // Date
+                HStack(alignment: .center, spacing: 4) {
+                    Assets.Icons.accessTime.swiftUIImage
+                        .frame(width: 16, height: 16)
+
+                    Text(sideEvent.timeText.currentLangTitle)
+                        .font(.system(size: 12, weight: .medium))
+                }
+            }
+            .foregroundStyle(AssetColors.Surface.onSurface.swiftUIColor)
+
+            Spacer(minLength: 0)
+
+            // Image
+            if let imageLink = sideEvent.link {
+                CacheAsyncImage(url: URL(string: imageLink)!) { image in
+                    image.resizable()
+                } placeholder: {
+                    Color.gray
+                }
+                .frame(width: 88, height: 88)
+                .scaledToFill()
+                .clipShape(RoundedRectangle(cornerRadius: 12))
+                .overlay(
+                    RoundedRectangle(cornerRadius: 12)
+                        .stroke(AssetColors.Outline.outline.swiftUIColor, lineWidth: 1)
+                )
+            } else {
+                Color.gray
+                    .frame(width: 88, height: 88)
+                    .clipShape(RoundedRectangle(cornerRadius: 12))
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 12)
+                            .stroke(AssetColors.Outline.outline.swiftUIColor, lineWidth: 1)
+                    )
+            }
+        }
+    }
+}
+
+private extension SideEvent {
+    var icon: Image {
+        switch self.mark {
+        case .favorite: return Assets.Icons.favorite.swiftUIImage
+        default: return Assets.Icons.favorite.swiftUIImage
```

Comment:
If I wouldn't use default case, I get compile error.
Is there better way, please?

Reply:
Hmm... There is a possibility that it is unavoidable when using KMP enum 😢 

I'll look for solutions about this 🙏🏼

Reply:
It looks difficult to remove the default case since Kotlin's enums are converted into Swift's class and Objective-C's @/interface... 😵‍💫
https://kotlinlang.org/docs/native-objc-interop.html#mappings

Reply:
Oh, It sounds difficult 👀
Thanks for letting me know! 🙏